awx/main/managers.py

```diff
@@ -145,7 +145,14 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type=None):
                     return (True, instance)
                 else:
                     return (False, instance)
-            instance = self.create(uuid=uuid, hostname=hostname, ip_address=ip_address, capacity=0, node_type=node_type)
+
+            # Create new instance, and fill in default values
+            create_defaults = dict(capacity=0, uuid=uuid)
+            if defaults is not None:
+                create_defaults.update(defaults)
+            if node_type == 'execution' and 'version' not in create_defaults:
+                create_defaults['version'] = RECEPTOR_PENDING
+            instance = self.create(hostname=hostname, ip_address=ip_address, node_type=node_type, **create_defaults)
```

Comment:
@rebeccahhh there's some refactoring of the default values here - meaning values that are set when creating a new instance versus getting an old instance.

I got bit debugging this, because the task code we had created confusion about whether capacity was set to 0, or node_type was set to execution. Then it went in loops checking capacity perpetually.

I left it with `node_type` being something that it will update existing instances with, but for execution nodes, it will plugin in "ansible-runner-???" for nodes registered by `awx-manage provision_instance`. I think this sounds right because when the installer runs that we don't know the capacity, and we have to wait for the mesh to be alive and working before we can get it.